src/charm_helpers.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper functions for the alertmanager charm."""
+from typing import List, Union
+from urllib.parse import urlparse
+
+
+def get_hostname_from_address(address: str) -> str:
```

Comment:
If `None`s are likely, I think we'd want to raise on them rather than filter them.  But looking at this again, I don't think they're likely (`None` happens if we `urlparse` on a netloc (eg: on something without a scheme)).  So what do you think of this?  

We can also filter on `None`, but I'm not sure we want to ignore `None` if it happens.  That feels like a sign of something badly wrong